src/libraries/System.Formats.Nrbf/src/System/Formats/Nrbf/SystemClassWithMembersAndTypesRecord.cs

```diff
@@ -77,10 +77,10 @@ internal SerializationRecord TryToMapToUserFriendly()
         }
         else if (MemberValues.Count == 2
             && HasMember("ticks") && HasMember("dateData")
-            && MemberValues[0] is long value && MemberValues[1] is ulong
+            && MemberValues[0] is long && MemberValues[1] is ulong dateData
```

Comment:
Is it safe to depend on the order of the fields in the payload? In other words, is the exact order of fields part of the BF contract for given type?

Reply:
`SerializationInfo` doesn't officially document the order, but in practice it enumerates elements in the same order in which they're added, and some types are sensitive to this ordering. It's akin to how if a dictionary / hashtable changes the order of enumeration or if a sort routine changes the relative order of "equal" elements, things break.

Reply:
I meant it in the connection with your other comment: Is the order considered a documented .NET Framework detail that it is ok to depend on; or is the order undocumented .NET Framework detail and we should not depend on it?

(My hunch would be that it should be the later.)

Reply:
It was my mistake to rely on the order of fields, I am going to push a fix in a minute.